weave-js/src/components/PagePanelComponents/Home/Browse3/filters/FilterBar.tsx

```diff
@@ -143,29 +152,36 @@ export const FilterBar = ({
     (field: string) => {
       const defaultOperator = getOperatorOptions(field)[0].value;
       const newModel = {
-        ...filterModel,
+        ...localFilterModel,
         items: [
-          ...filterModel.items,
+          ...localFilterModel.items,
           {
-            id: filterModel.items.length,
+            id: localFilterModel.items.length,
             field,
             operator: defaultOperator,
           },
         ],
       };
-      setFilterModel(newModel);
+      setLocalFilterModel(newModel);
     },
-    [filterModel, setFilterModel]
+    [localFilterModel]
+  );
+
+  const debouncedSetFilterModel = useMemo(
+    () =>
+      _.debounce((newModel: GridFilterModel) => setFilterModel(newModel), 300),
+    [setFilterModel]
   );
 
   const onUpdateFilter = useCallback(
     (item: GridFilterItem) => {
-      const oldItems = filterModel.items;
+      const oldItems = localFilterModel.items;
       const index = oldItems.findIndex(f => f.id === item.id);
 
       if (index === -1) {
-        const newModel2 = {...filterModel, items: [item]};
-        setFilterModel(newModel2);
+        const newModel2 = {...localFilterModel, items: [item]};
+        setLocalFilterModel(newModel2);
```

Comment:
i would instead create a callback: `updateFilterModel` which internally sets the "local" working copy AND calls the debounced one. Seems weird to have the dual call here

Reply:
I agree this feels wrong, but it would only ever be used in this one place, the other times we `setLocalFilterModel` we want to either _not_ update the original filter or directly set it instead of using the debounced. 

I will try to appease the style gods while not being verbose